### ISSUE TITLE
Uncaught TypeError: Cannot read property 'start' of undefined

### ISSUE THREAD

**Issue by asampal** [jasongilman/proto-repl#83]
1. Open up a REPL while not in a project
2. Try to start typing in the entry area of the REPL

This was also happening in 1.1.2

**Atom Version**: 1.7.0-beta0
**System**: Microsoft Windows 10 Pro
**Thrown From**: [proto-repl](https://github.com/jasongilman/proto-repl) package, v1.1.3
### Stack Trace

Uncaught TypeError: Cannot read property 'start' of undefined

```
At /C:/Users/adi/.atom/packages/proto-repl/lib/repl-text-editor.coffee:118

TypeError: Cannot read property 'start' of undefined
    at ReplTextEditor.module.exports.ReplTextEditor.allowsRangeChange (file:///C:/Users/adi/.atom/packages/proto-repl/lib/repl-text-editor.coffee:114:10)
    at ReplTextEditor.module.exports.ReplTextEditor.allowsChange (file:///C:/Users/adi/.atom/packages/proto-repl/lib/repl-text-editor.coffee:118:26)
    at file:///C:/Users/adi/.atom/packages/proto-repl/lib/repl-text-editor.coffee:124:37
    at TextBuffer.textEditor.buffer.applyChange (file:///C:/Users/adi/.atom/packages/proto-repl/lib/repl-text-editor.coffee:129:10)
    at TextBuffer.module.exports.TextBuffer.revertToCheckpoint (C:\Users\adi\AppData\Local\atom\app-1.7.0-beta0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:865:16)
    at TextBuffer.module.exports.TextBuffer.transact (C:\Users\adi\AppData\Local\atom\app-1.7.0-beta0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:831:14)
    at TextEditor.module.exports.TextEditor.transact (C:\Users\adi\AppData\Local\atom\app-1.7.0-beta0\resources\app.asar\src\text-editor.js:1274:26)
    at C:\Users\adi\AppData\Local\atom\app-1.7.0-beta0\resources\app.asar\src\text-editor.js:868:24
    at TextEditor.module.exports.TextEditor.mergeSelections (C:\Users\adi\AppData\Local\atom\app-1.7.0-beta0\resources\app.asar\src\text-editor.js:2166:43)
    at TextEditor.module.exports.TextEditor.mergeIntersectingSelections (C:\Users\adi\AppData\Local\atom\app-1.7.0-beta0\resources\app.asar\src\text-editor.js:2132:35)
    at TextEditor.module.exports.TextEditor.mutateSelectedText (C:\Users\adi\AppData\Local\atom\app-1.7.0-beta0\resources\app.asar\src\text-editor.js:866:19)
    at TextEditor.module.exports.TextEditor.insertText (C:\Users\adi\AppData\Local\atom\app-1.7.0-beta0\resources\app.asar\src\text-editor.js:832:19)
    at TextEditor.object.(anonymous function) [as insertText] (C:\Users\adi\AppData\Local\atom\app-1.7.0-beta0\resources\app.asar\node_modules\underscore-plus\lib\underscore-plus.js:77:27)
    at TextEditorComponent.module.exports.TextEditorComponent.onTextInput (C:\Users\adi\AppData\Local\atom\app-1.7.0-beta0\resources\app.asar\src\text-editor-component.js:474:35)
    at HTMLDivElement.<anonymous> (C:\Users\adi\AppData\Local\atom\app-1.7.0-beta0\resources\app.asar\src\text-editor-component.js:3:61)
```
### Commands

```
     -3:02.9.0 settings-view:check-for-package-updates (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
     -1:32.2.0 proto-repl:toggle (atom-text-editor.editor.is-focused)
```
### Config

``` json
{
  "core": {
    "autoHideMenuBar": true
  },
  "proto-repl": {
    "leinPath": "c:\\utils\\"
  }
}
```
### Installed Packages

``` coffee
# User
atom-yeoman, v0.3.15
ink, v0.4.1
ionide-fake, v1.2.2
ionide-fsharp, v1.7.0
ionide-fsi, v1.1.2
ionide-installer, v1.3.0
ionide-paket, v2.2.6
ionide-webview, v1.0.3
ionide-yeoman, v1.0.2
language-elm, v1.3.3
linter, v1.11.4
linter-elm-make, v0.4.0
lisp-paredit, v0.5.2
minimap, v4.21.0
minimap-hide, v0.3.0
Parinfer, v1.14.0
proto-repl, v1.1.3
proto-repl-charts, v0.3.2
tool-bar, v0.4.0

# Dev
No dev packages
```

**Comment by jasongilman** [comment]
Atom has changed their text buffer code in 1.7.0. Thanks for reporting this problem. I released 1.1.5 with a change that should fix this. Can you let me know if this resolves the issue?

**Comment by asampal** [comment]
Seems fine now, @jasongilman - thanks. One thing to note in case others have problems is that after updating to 1.1.5, which looked to installed with no problems, when I went to toggle the REPL I was still getting this error. I had to re-start Atom to get it working. I suppose for some plugins Atom must be re-started, but it didn't give any indication of that and when I checked the settings page, it showed 1.1.5 as the current version.
